internal/kubernetes/client.go

```diff
@@ -66,20 +65,37 @@ func createControllerRuntimeClient() (client.Client, error) {
 	return clnt, err
 }
 
-// FetchEnterpriseContractPolicy gets the Enterprise Contract Policy from the given namespace in a Kubernetes cluster
-func (k *Client) FetchEnterpriseContractPolicy(ctx context.Context, name types.NamespacedName) (*ecp.EnterpriseContractPolicy, error) {
+// FetchEnterpriseContractPolicy gets the Enterprise Contract Policy from the given
+// reference in a Kubernetes cluster.
+//
+// The reference is expected to be in the format [<namespace>/]<name>. If it does not contain
+// a namespace, the current namespace is used.
+func (k *Client) FetchEnterpriseContractPolicy(ctx context.Context, ref string) (*ecp.EnterpriseContractPolicy, error) {
+	if len(ref) == 0 {
+		return nil, errors.New("policy reference cannot be empty")
+	}
+	log.Debugf("Raw policy reference: %q", ref)
+
+	name, err := NamespacedName(ref)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("Parsed policy reference: %v", name)
 	if name.Namespace == "" {
-		return nil, errors.New("missing namespace")
+		return nil, errors.New("unable to determine namespace for policy")
 	}
 
 	policy := &ecp.EnterpriseContractPolicy{}
-
-	err := k.client.Get(ctx, name, policy)
-	if err != nil {
-		log.Debug("Failed to get policy from cluster!")
+	if err := k.client.Get(ctx, *name, policy); err != nil {
+		log.Debugf("Failed to fetch the policy from cluster: %s", err)
 		return nil, err
 	}
-	policyJson, _ := json.Marshal(policy.Spec)
-	log.Debugf("Policy fetched:\n%s", policyJson)
+	if policyJson, err := json.Marshal(policy.Spec); err != nil {
+		log.Debugf("Unable to marshal policy to json: %s", policyJson)
```

Comment:
Do you need to return the err after logging?

Reply:
Well I don't know... The call to `json.Marshall` is only meant to assist in logging (which seems a bit odd to me). In the previous code, the error code was ignored completely. My assumption is that it was intended to be an optional thing. I can't picture a scenario where the `json.Marshall` call actually fails though. I can switch back to ignoring the error as it was before?

Reply:
you're right, I didn't look close enough. Maybe log `err` though instead of `policyJson`?

Reply:
I took it a step further. I removed the `json.Marshall` call altogether. Instead, I just log the policy object directly. I'm using the `%#v` "verb" now so it actually shows up in a more readable way. Wdyt?